examples/translation_example.py

```diff
@@ -148,53 +162,231 @@ def parse_arguments():
+        else:
+            break
+    while len(end):
+        if end[0] in TERMINAL_PUNCTUATIONS:
+            mid += end[0]
+            end = end[1:]
+        else:
+            break
+    return pd.Series([start, mid, end], index=['start_sym', 'indic_proc_text', 'end_sym'])
+
+def has_alphabet_characters(text):
+    return any(c.isalpha() for c in text)
+
+def sent_tokenizer(line):
+    return sent_tokenize(line)
+
+def custom_tokenize(text):
+    split_text = re.split(r"(\#{2,}|\_{2,}|\…{2,}|\+{2,}|\.{2,}|\-{3,}|\*{2,}|\~{2,}|\={2,}|\!{2,}|\n|\t|\‣|\⁃|\⁌|\⁍|\●|\○|\•|\·|\◘|\◦|\⦾|\⦿|\|)", text)
+    split_text = [s for s in split_text if len(s) > 0]
+    tokenized_sentences = []
+    len_flag = False
+    for line in split_text:
+        # Tokenize sentences using NLTK's sent_tokenize function
+        if has_alphabet_characters(line) == True:
+            sentences = sent_tokenizer(line)
+            i = 0
+            j = 0
+            tokenized_snt_lst = []
+            str = ""
+            while i < len(line):
+                ch = line[i]
+                if j < len(sentences):
+                    stripped_sent = sentences[j].strip()
+                    if len(stripped_sent) == 0:
+                        j += 1
+                        continue
+                    if ch == stripped_sent[0]:
+                        if str != "":
+                            tokenized_snt_lst.append(str)
+                        tokenized_snt_lst.append(stripped_sent)
+                        i += len(stripped_sent)
+                        j += 1
+                        str = ""
+                    else:
+                        str += ch
+                        i += 1
+                else:
+                    str += ch
+                    i += 1
+            if str != "":
+                tokenized_snt_lst.append(str)
+            # Add the tokenized sentences to the list
+            tokenized_sentences.extend(tokenized_snt_lst)
+        else:
+            tokenized_sentences.append(line)
+
+    tokenized_sentence_len = []
+    for sentence in tokenized_sentences:
+        sent = sentence.split()
+        # removing the sentences with word length greater than threshold as the model may not be able translate it due to constraint on output token size
+        if len(sent) <= MAX_TOKEN_LEN:
+            tokenized_sentence_len.append(sentence)
+        else:
+            pass
+    return tokenized_sentence_len
+
+
+def process_input_text(ddf):
+    ddf = ddf.to_pandas()
+    ddf['indic_proc_text'] = ddf['text'].apply(custom_tokenize)
+    ddf['doc_id'] = range(1, len(ddf) + 1)
+    ddf = ddf.explode('indic_proc_text')
+    ddf = ddf.reset_index(drop=True)
+    ddf=ddf.drop('text', axis=1)
+    ddf[['start_sym', 'indic_proc_text', 'end_sym']] = ddf['indic_proc_text'].apply(truncate_start_end_symbols)#, result_type='expand')
+    ddf = cudf.DataFrame.from_pandas(ddf)
+    return ddf
+
+def concat_strings(series):
+    return ''.join(series)
```

Comment:
Is this applied on string series or python list ?

Reply:
This is applied on string series.